patch configure.in to correctly check multiple libpng soname versions
the resulting patch to configure is in ../configure-libpng.patch
see https://trac.sagemath.org/ticket/25818
--- a/configure.in	2018-07-10 18:26:56.330278000 +0000
+++ b/configure.in	2018-07-10 19:06:58.602278000 +0000
@@ -100,7 +100,7 @@
 	[ if test "x$enableval" = "xno"; then CONFIG_PNG="no"; fi], [])
   
 if test "x$CONFIG_PNG" = "xyes"; then 
-   AC_CHECK_HEADERS(png.h, AC_CHECK_LIB(png,main)) dnl AC_CHECK_LIBS(main,[png16,png14,png12,png]))
+   AC_CHECK_HEADERS(png.h, AC_SEARCH_LIBS(png_sig_cmp,[png16 png14 png12 png]))
 fi
